lib/Conversion/ConvertToArcs/ConvertToArcs.cpp

```diff
@@ -294,10 +316,12 @@ void Converter::absorbRegs(HWModuleOp module) {
     ++numTrivialRegs;
 
     // Set the arc's clock to the clock of the registers we've absorbed, bump
-    // the latency up by one to account for the registers, and update the output
-    // names. Then replace the registers.
+    // the latency up by one to account for the registers, add the reset if
+    // present and update the output names. Then replace the registers.
     arc.getClockMutable().assign(clock);
     arc.setLatency(arc.getLatency() + 1);
+    if (reset)
+      arc.getResetMutable().assign(reset);
```

Comment:
What if there is already a reset present?

Reply:
Are you happy for this just to be covered with an assertion? I suppose we could `comb.or` multiple reset lines, but I'm unable to think of a situation where you would expect/desire a reset to already be present on the StateOp at this point, unless there are additions you have planned.

Reply:
The situation/assertion could be triggered with an input that has both state ops and compreg ops in there which does not happen currently. But the fact that a user can construct such an IR is enough of a reason that we shouldn't use an assertion. However, I think it's not necessary to support this case right now, so I would just go for an `emitError` in that case as well.

Reply:
Ah of course! Sounds good, I'll do that :)